src/SignalR/clients/csharp/Http.Connections.Client/src/WebSocketConnectionContext.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+
+namespace Microsoft.AspNetCore.Http.Connections.Client
+{
+    /// <summary>
+    /// Used to make a connection to an SignalR using a WebSocket-based transport.
+    /// </summary>
+    public sealed class WebSocketConnectionContext
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WebSocketConnectionContext"/> class.
+        /// </summary>
+        /// <param name="uri">The URL to connect to.</param>
+        /// <param name="options">The connection options to use.</param>
+        public WebSocketConnectionContext(Uri uri, HttpConnectionOptions options)
+        {
+            Uri = uri;
+            Options = options;
+        }
+
+        /// <summary>
+        /// Gets the URL to connect to.
+        /// </summary>
+        public Uri Uri { get; }
```

Comment:
Was there an issue using `init` properties? https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-9.0/init

Reply:
yeah, when i was using 
```
        public Uri Uri { get; init; }
```

it gave
```
4>------ Build started: Project: Microsoft.AspNetCore.Http.Connections.Client, Configuration: Debug Any CPU ------
4>...aspnetcore\src\SignalR\clients\csharp\Http.Connections.Client\src\WebSocketConnectionContext.cs(27,31,27,35): error CS0518: Predefined type 'System.Runtime.CompilerServices.IsExternalInit' is not defined or imported
4>...aspnetcore\src\SignalR\clients\csharp\Http.Connections.Client\src\WebSocketConnectionContext.cs(32,53,32,57): error CS0518: Predefined type 'System.Runtime.CompilerServices.IsExternalInit' is not defined or imported
4>Done building project "Microsoft.AspNetCore.Http.Connections.Client.csproj" -- FAILED.
```

i have VS 2019 16.9.4. the same happens when i run `./build.cmd -all` in the SignalR folder. `build.cmd` uses `6.0.0-preview.5.21229.9` version. I see that class was added at Jun 2020 to runtime and should be part of 5.0 and not sure why it fails.

Reply:
https://stackoverflow.com/questions/64749385/predefined-type-system-runtime-compilerservices-isexternalinit-is-not-defined

Looks like we would need to add the type ourselves for netstandard2.0, netstandard2.1, and net461

Reply:
it makes sense. i followed example and added it. might need guidance would it be ok or not. as if it would be used in other projects it might have sense to place it to `$(SharedSourceRoot)IsExternalInit.cs` so other projects can reference it.

Also i just used preprocessor directives but other way might be to specify the same constraints on csproj side and i'm not sure what is prefered way.

Also as another point to simplify it, it can be updated to `#ifdef !NET5_0_OR_GREATER` so it would be working for everything that is lower than 5_0 but it would be less specific in that case

Reply:
We can put it in https://github.com/dotnet/aspnetcore/tree/main/src/Shared

> it can be updated to `#ifdef !NET5_0_OR_GREATER`

That sounds good.

Did you forget to include the file when pushing?

Reply:
yeah, i did :) i'll move that file to shared place and will update it to use !NET5_0 one

Reply:
pushed updated file to Shared folder and updated csproj file to point to it